packages/drawer/index.js

```diff
@@ -0,0 +1,218 @@
+import DrawerContent from './Content';
+import DrawerSubtitle from './Subtitle';
+import DrawerTitle from './Title';
+import DrawerAppContent from './AppContent';
+import FocusTrap from 'focus-trap-react';
+
+const {cssClasses: listCssClasses} = MDCListFoundation;
+
+class Drawer extends React.Component {
+  previousFocus_ = null;
+  foundation_ = null;
+  drawerElement_ = React.createRef();
+
+  state = {classList: new Set()};
+
+  componentDidMount() {
+    const {dismissible, modal, open} = this.props;
+    if (dismissible) {
+      this.foundation_ = new MDCDismissibleDrawerFoundation(this.adapter);
+      this.foundation_.init();
+    } else if (modal) {
+      this.foundation_ = new MDCModalDrawerFoundation(this.adapter);
+      this.foundation_.init();
+    }
+
+    if (open && this.foundation_) {
+      this.foundation_.open();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {dismissible, modal, open} = this.props;
+    if (!(dismissible || modal)) return;
+
+    if (open !== prevProps.open) {
+      open ? this.foundation_.open() : this.foundation_.close();
+    }
+  }
+
+  componentWillUnmount() {
+    if (!this.foundation_) return;
+    this.foundation_.destroy();
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className, dismissible, modal} = this.props;
+    return classnames('mdc-drawer',
+      Array.from(classList),
+      className, {
+        'mdc-drawer--dismissible': dismissible,
+        'mdc-drawer--modal': modal,
+      }
+    );
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className) => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className) => this.classes.split(' ').includes(className),
+      elementHasClass: (element, className) => element.classList.contains(className),
+      saveFocus: () => {
+        this.previousFocus_ = document.activeElement;
+      },
+      restoreFocus: () => {
+        const hasPreviousFocus = this.previousFocus_ && this.previousFocus_.focus;
+        const drawerElement = this.drawerElement_ && this.drawerElement_.current;
+        if (drawerElement && hasPreviousFocus && drawerElement.contains(document.activeElement)) {
+          this.previousFocus_.focus();
+        }
+      },
+      focusActiveNavigationItem: () => {
+        const drawerElement = this.drawerElement_ && this.drawerElement_.current;
+        if (!drawerElement) return;
+        const activeNavItemEl = drawerElement.querySelector(`.${listCssClasses.LIST_ITEM_ACTIVATED_CLASS}`);
+        if (activeNavItemEl) {
+          activeNavItemEl.focus();
+        }
+      },
+      notifyClose: this.props.onClose,
+      notifyOpen: this.props.onOpen,
+      trapFocus: () => this.setState({activeTrap: true}),
```

Comment:
I'm actually using this documentation https://github.com/davidtheclark/focus-trap-react#usage

It made sense to me since we're activating the focus trap. Maybe instead we can call it `activeFocusTrap`?